components/eamxx/src/physics/ml_correction/eamxx_ml_correction_process_interface.cpp

```diff
@@ -22,40 +28,90 @@ void MLCorrection::set_grids(
   // Nevertheless, for output reasons, we like to see 'kg/kg'.
   auto Q = kg / kg;
   Q.set_string("kg/kg");
-
+  constexpr int ps = SCREAM_SMALL_PACK_SIZE;
   m_grid                = grids_manager->get_grid("Physics");
   const auto &grid_name = m_grid->name();
   m_num_cols = m_grid->get_num_local_dofs();  // Number of columns on this rank
   m_num_levs =
       m_grid->get_num_vertical_levels();  // Number of levels per column
-
   // Define the different field layouts that will be used for this process
 
   // Layout for 3D (2d horiz X 1d vertical) variable defined at mid-level and
   // interfaces
   FieldLayout scalar3d_layout_mid{{COL, LEV}, {m_num_cols, m_num_levs}};
 
-  // Set of fields used strictly as input
-  add_field<Required>("qv", scalar3d_layout_mid, Q, grid_name, "tracers");
-
-  // Set of fields used strictly as output
-  add_field<Computed>("qv_nudging_tend", scalar3d_layout_mid, Q, grid_name);
-
-  // Set of fields used as input and output
-  // - There are no fields used as both input and output.
+  /* ----------------------- WARNING --------------------------------*/
+  /* The following is a HACK to get things moving, we don't want to
+   * add all fields as "updated" long-term.  A separate stream of work
+   * is adapting the infrastructure to allow for a generic "add_field" call
+   * to be used here which we can then setup using the m_fields_ml_output_variables variable
+   */
+  add_field<Updated>("T_mid", scalar3d_layout_mid, K, grid_name, ps);
+  add_field<Updated>("qv",    scalar3d_layout_mid, Q, grid_name, "tracers", ps);
+  add_field<Updated>("u",     scalar3d_layout_mid, m/s, grid_name, ps);
```

Comment:
This may not work as you indended. EAMxx can only create the u/v fields _after_ all fields have been created, by slicing the `horiz_wind` vector field. But it calls them `U` and `V` (uppercase). The two fields you declare here are two separate fields, independent of `horiz_winds` (or `U` and `V`). However, it would still not work if you used `U` and `V`, with the AD that will crap out (I can explain why offline).

If you want to grab the two components of horiz_winds separately, you need to request `horiz_winds`, and then simply grab the two components in `run_impl`:
```
const auto& u = get_field_out("horiz_winds").get_component(0);
const auto& v = get_field_out("horiz_winds").get_component(1);
```

Reply:
Okay, I was following what Aaron did in this [PR](https://github.com/E3SM-Project/scream/pull/2378), but admittedly I haven't actually tried ML correcting `u` and `v`. Maybe this explains why the nudging tendencies for `u` and `v` seem off. Pinging you and Aaron offline for this.